### PR TITLE
Update CODING_STYLE

### PR DESCRIPTION
**Pull request by chfast**
- Renamed CodingStandards.txt to CODING_STYLE.md
- Converted to Markdown.
- Removed some formatting related items.
- Added 2 notes about C++ Core Guidelines (see TODO comments).

### CODE REVIEW

File: CODING_STYLE.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
+# C++ Ethereum Coding Style
+
+## Code Formatting
+
+Use clang-format tool to format your changes, see [CONTRIBUTING](CONTRIBUTING.md) for details.
+
+
+## Namespaces
+
+a. No "using namespace" declarations in header files.
```
Comment (gumb0) [inline CODING_STYLE.md]: This a-b-c-list is not rendered correctly by GitHub, use 1-2-3-list

Comment (chfast) [reply CODING_STYLE.md]: I missed this one.

File: CODING_STYLE.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+3. All preprocessor symbols (macros, macro arguments) in full uppercase with underscore word separation.
+
+
+All other entities' first alpha is lower case.
+
+
+## Variable prefixes
+
+1. Leading underscore "_" to parameter names.
+   - Exception: "o_parameterName" when it is used exclusively for output. See 6(f).
```
Comment (gumb0) [inline CODING_STYLE.md]: It's 6.5 now

File: CODING_STYLE.md [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+
+## Declarations
+
+1. {Typename} + {qualifiers} + {name}. (**TODO**: Against [NL.26](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#nl26-use-conventional-const-notation))
+2. Only one per line.
+3. Favour declarations close to use; don't habitually declare at top of scope ala C.
+4. Always pass non-trivial parameters with a const& suffix.
+5. If a function returns multiple values, use std::tuple (std::pair acceptable).
+   Prefer not using */& arguments, except where efficiency requires.
+   (**TODO**: Compare with [C++ Core Guidelines F.21](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#f21-to-return-multiple-out-values-prefer-returning-a-tuple-or-struct) )
```
Comment (gumb0) [inline CODING_STYLE.md]: I think we agree overall with Core Guidelines here, so maybe put just `See also [C++ Core Guidelines F.21]`

File: CODING_STYLE.md [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+6. Never use a macro where adequate non-preprocessor C++ can be written.
+7. Make use of auto whenever type is clear or unimportant:
+   - Always avoid doubly-stating the type.
+   - Use to avoid vast and unimportant type declarations.
+   - However, avoid using auto where type is not immediately obvious from the context, and especially not for arithmetic expressions.
+8. Don't pass bools: prefer enumerations instead.
+9. Prefer enum class to straight enum.
+
+
+    // WRONG:
```
Comment (gumb0) [inline CODING_STYLE.md]: This code is not rendered as code

File: CODING_STYLE.md [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+6. For a property 'foo'
+   - Member: m_foo;
+   - Getter: foo(); also: for booleans, isFoo()
+   - Setter: setFoo();
+
+
+## Naming
+
+1. Collection conventions:
+   - -s means std::vector e.g. using MyTypes = std::vector<MyType>
```
Comment (gumb0) [inline CODING_STYLE.md]: put these things in backticks, otherwise names inside angle brackets are not rendered

File: CODING_STYLE.md [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+   - Spell correctly.
+   - Think carefully about the class's purpose.
+   - Imagine it as an isolated component to try to decontextualise it when considering its name.
+   - Don't be trapped into naming it (purely) in terms of its implementation.
+
+
+
+## Type-definitions
+
+1. Prefer 'using' to 'typedef'. e.g. using ints = std::vector<int>; rather than typedef std::vector<int> ints;
```
Comment (gumb0) [inline CODING_STYLE.md]: here too, names in angle brackets are not visible

File: CODING_STYLE.md [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+6. For a property 'foo'
+   - Member: m_foo;
+   - Getter: foo(); also: for booleans, isFoo()
+   - Setter: setFoo();
+
+
+## Naming
+
+1. Collection conventions:
+   - ~s means `std::vector` e.g. `using MyTypes = std::vector<MyType>`
```
Comment (gumb0) [inline CODING_STYLE.md]: Why did you change to from minus to tildas here?
